### DISCUSSION TITLE
Make the plugin work with OpenBao too?

### DISCUSSION THREAD

**Discussion by mh96794** [jfrog/vault-plugin-secrets-artifactory#184] [General]
Would it take much to make the vault-plugin-secrets-artifactory plugin to work with the OpenBao fork of Vault too https://openbao.org ? 

At the moment it does not work with OpenBao. 
After successful plugin registration the "enable" command fails with an "invalid backend" error, e.g.

$ ./bao secrets enable -path=ausartifactory artifactory-secrets-plugin_1.6.1dev_linux_amd64
Error enabling: Error making API request.
URL: POST http://127.0.0.1:8200/v1/sys/mounts/ausartifactory
Code: 400. Errors:
* invalid backend version: 2 errors occurred:
	* Unrecognized remote plugin message:
 This usually means ...

**Comment by alexhung** [comment]
@mh96794 Thanks for the question. I have not investigate what's required to make the plugin compatible with OpenBao. I'll add this to our plan.

**Reply by mh96794** [reply]
That is great. Thank you for considering this.

**Reply by alexhung** [reply]
@mh96794 FYI this is caused by OpenBao (https://github.com/orgs/openbao/discussions/317) which they have a fix coming.

**Reply by mh96794** [reply]
@alexhung  Excellent. Thank you for that pointer.

**Reply by alexhung** [reply]
@mh96794 The issue has been fixed a while ago. I've just updated the makefile and GitHub workflow to execute the acceptance tests with both Vault and OpenBao cli.
